packages/ts/src/components/treemap/index.ts

```diff
@@ -0,0 +1,304 @@
+      .style('cursor', config.showTileClickAffordance ? d => !d.children ? 'pointer' : null : null)
+
+
+    const mergedTiles = tiles.merge(tilesEnter)
+
+    smartTransition(mergedTiles.select(`rect.${s.tile}`), duration)
+      .style('fill', d => d._fill ?? getColor(d, config.tileColor))
+      .style('opacity', 1)
+      .attr('x', d => d.x0)
+      .attr('y', d => d.y0)
+      .attr('width', d => d.x1 - d.x0)
+      .attr('height', d => d.y1 - d.y0)
+
+    // Update clipPath rects
+    let svg: Element | null = this.g.node()
+    while (svg && !(svg instanceof SVGSVGElement)) svg = svg.parentElement
+    const defs = svg ? (select(svg).select('defs').empty() ? select(svg).append('defs') : select(svg).select('defs')) : null
+    if (!defs) return
+    const defsSelection = (defs as Selection<SVGDefsElement, unknown, null, undefined>)
+    const clipPaths = defsSelection.selectAll<SVGClipPathElement, TreemapNode<Datum>>('clipPath')
+      .data(visibleNodes, (d: TreemapNode<Datum>) => d._id)
+
+    clipPaths.enter()
+      .append('clipPath')
+      .attr('id', (d: TreemapNode<Datum>) => `clip-${d._id}`)
+      .append('rect')
+      .attr('x', (d: TreemapNode<Datum>) => d.x0)
+      .attr('y', (d: TreemapNode<Datum>) => d.y0)
+      .attr('width', (d: TreemapNode<Datum>) => Math.max(0.1, d.x1 - d.x0))
+      .attr('height', (d: TreemapNode<Datum>) => Math.max(0.1, d.y1 - d.y0))
+      .attr('rx', rx)
+      .attr('ry', rx)
+
+    clipPaths.exit().remove()
+
+    tilesEnter
+      .append('g')
+      .attr('class', 'label-group')
+      .append('text')
+      .attr('class', s.label)
+      .style('opacity', 0)
+
+    mergedTiles.select('g.label-group text')
+      .text(d => `${d.data.key}: ${numberFormat(d.value)}`)
+      .style('font-size', function (d) {
+        const sqrtVal = Math.sqrt(d.value ?? 0)
+        return config.enableTileLabelFontSizeVariation && !d.children
+          ? `${fontSizeScale(sqrtVal)}px`
+          : `${fontSizeScale.range()[1]}px`
+      })
+      .attr('dominant-baseline', 'middle')
+      .each((_, i, nodes) => {
+        const text = select(nodes[i] as SVGTextElement)
+        text.selectAll('tspan').attr('dominant-baseline', 'middle')
+      })
+
+    // Transition group position
+    smartTransition(mergedTiles.select('g.label-group'), duration)
+      .attr('transform', d => `translate(${d.x0 + config.labelOffsetX},${d.y0 + config.labelOffsetY})`)
+
+    // Transition text opacity only (fade-in)
+    smartTransition(mergedTiles.select('g.label-group text'), duration)
+      .style('opacity', 1)
+
+    // Hide labels that don't meet criteria
+    mergedTiles.select(`text.${s.label}`)
+      .style('display', d => {
+        const isAllowedNode = config.labelInternalNodes ? true : !d.children
+        return isAllowedNode && this._isTileLargeEnough(d) ? null : 'none'
+      })
+      // Make the internal labels semibold
+      .style('font-weight', d => d.children ? '500' : 'normal')
```

Comment:
I would do this via assigning a class name and having a CSS configurable variable